### PR TITLE
[NPU] Support shared expert dual stream optimization

### PR DESCRIPTION
**Pull request by iridiumine**
## Motivation

On NPU, the shared expert and routed expert in MoE models are executed sequentially on the same stream, leaving compute resources underutilized. This PR introduces a dual stream optimization that overlaps shared expert computation with routed expert computation on independent NPU streams.

## Modifications

- Added `shared_expert_on_independent_stream` in `cmo.py` to offload shared expert computation to independent NPU streams with dedicated core allocation (8 cores).
- Modified `Qwen2MoeModel._forward_deepep` to execute shared expert and routed expert on separate streams in parallel when `SGLANG_NPU_USE_MULTI_STREAM` is enabled during CUDA graph mode.

## Accuracy Tests

```
# shared expert stream
export SGLANG_NPU_USE_MULTI_STREAM=1

MODEL_PATH=/home/weights/Qwen3.6-35B-A3B
python3 -m sglang.launch_server \
        --model-path ${MODEL_PATH} \
        --attention-backend ascend \
        --device npu \
        --tp-size 4 --nnodes 1 --node-rank 0 \
        --chunked-prefill-size -1 --max-prefill-tokens 16384 \
        --trust-remote-code \
        --host 127.0.0.1 --max-running-requests 128 \
        --mem-fraction-static 0.8 \
        --port 9903 \
        --mm-attention-backend ascend_attn --max-total-tokens 800000 \
        --dtype bfloat16 --mamba-ssm-dtype bfloat16 \
        --base-gpu-id 4 \
        --enable-multimodal \
        --disable-radix-cache \
        --dp-size 2 --enable-dp-attention --enable-dp-lm-head \
        --speculative-algorithm NEXTN --speculative-num-steps 3 --speculative-eagle-topk 1 --speculative-num-draft-tokens 4 \
        --speculative-draft-model-quantization unquant \
        --moe-a2a-backend deepep --deepep-mode auto \
        --cuda-graph-bs 8 12 16 20 24 32 40 48 56 64 \
```

```
+-----------------+-----------+----------+----------+-------+---------+---------+
| Model           | Dataset   | Metric   | Subset   |   Num |   Score | Cat.0   |
+=================+===========+==========+==========+=======+=========+=========+
| Qwen3.6-35B-A3B | gsm8k     | mean_acc | main     |   256 |  0.9766 | default |
+-----------------+-----------+----------+----------+-------+---------+---------+
+-----------------+-----------+----------+------------------------------------------+-------+---------+----------------+
| Qwen3.6-35B-A3B | ceval     | mean_acc | OVERALL                                  |  1346 |  0.893  | -              |
+-----------------+-----------+----------+------------------------------------------+-------+---------+----------------+
```

## Speed Tests and Profiling

Benchmarked on Qwen3.6-35B-A3B, 128 concurrent requests:

Metric | Dual Stream Disabled | This PR | Improvement
-- | -- | -- | --
throughput (req/s) | 1.72 | 1.91 | +11.0%
Output token throughput (tok/s) | 2580.73 | 2868.83 | +11.2%
Total token throughput (tok/s) | 8602.42 | 9562.76 | +11.2%
Mean E2E Latency (ms) | 57695 | 52973 | -8.2%
Mean TTFT (ms) | 14646 | 13708 | -6.4%
Mean TPOT (ms) | 28.72 | 26.19 | -8.8%

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

### CODE REVIEW

File: python/sglang/srt/models/qwen2_moe.py [hunk 1]
```diff
@@ -381,11 +390,21 @@ def _forward_shared_experts(self, hidden_states: torch.Tensor):
...
+            is_npu()
+            and envs.SGLANG_NPU_USE_MULTI_STREAM.get()
+            and forward_batch.forward_mode.is_cuda_graph()
+        )
         shared_output = None
         if hidden_states.shape[0] > 0:
             # router_logits: (num_tokens, n_experts)
             router_logits, _ = self.gate(hidden_states)
-            shared_output = self._forward_shared_experts(hidden_states)
+            if enable_dual_stream:
```
Comment (OrangeRedeng) [inline python/sglang/srt/models/qwen2_moe.py:402]: Does it only work for Qwen?

Comment (iridiumine) [reply python/sglang/srt/models/qwen2_moe.py:402]: I only tried the Qwen3.5 series models, but you can try other MoE models.
